src/lib/net.py

```diff
@@ -63,13 +62,14 @@ def download_snapshot(
 
 def download(url: str, file_handle: BinaryIO, progress: bool = False) -> None:
```

Comment:
Probably not worth considering for now, but longer term it might be worth making this an async function and extracting the progress bar behavior to a higher-level abstraction kind of like https://gist.github.com/khardix/e8b7e87f9e2e1c9c8e88e3774c52ec77.

Reply:
Sure. This is pretty heavily used across multiple data sources so a speedup could have significant effect on the total processing time.